### ISSUE TITLE
All unit/integration tests run sync over async

### ISSUE THREAD

**Issue by gliljas** [confluentinc/confluent-kafka-dotnet#1411]
Description
===========
1. It's a bit concerning with all the .Wait():s and .Result:s
2. It means tests are not really run in the way they would be used in "production"
3. Writing tests in a proper async fashion may influence design decisions for the actual API

I'll produce a PR if there's interest

**Comment by mhowlett** [comment]
fair point, though the most likely place bugs are likely to occur is not here. i've added this to a big document we have of additional testing we want to do. there are many todo items above this.
